modes/elixir-mode/elixir-mode.lisp

```diff
@@ -0,0 +1,107 @@
+  (:export :*elixir-mode-hook*
+           :elixir-mode))
+(in-package :lem-elixir-mode)
+
+(defvar *elixir-keywords*
+  '("case" "cond" "for" "if" "quote" "raise" "receive" "send" 
+    "super" "throw" "try" "unless" "unquote" "unquote_splicing" "with" 
+    
+    "import" "require" "use" "alias"
+    
+    "def" "defp" "defmodule" "defprotocol"
+    "defmacro" "defmacrop" "defdelegate"
+    "defexception" "defstruct" "defimpl"
+    "defguard" "defguardp" "defcallback"
+    "defoverridable" 
+    "end" "after" "else" "rescue" "catch" 
+
+    "not" "and" "or" "when" "in"
+
+    "fn" "do" "end" "after" "else" "rescue" "catch"
+
+    "->" "<-" "=>" "|>"))
+
+(defvar *elixir-boolean-literals*
+  '("true" "false"))
+
+(defvar *elixir-null-literal*
+  '("nil"))
+
+;; From:  http://elixir-lang.org/getting-started/basic-operators.html
+(defvar *elixir-operators*
+  '("+" "*" "++" "--" "<>" "||" "&&" "!"
+    "==" "!=" "===" "!==" ">=" "<="))
+
+
+(defun tokens (boundary strings)
+  (let ((alternation
+          `(:alternation ,@(sort (copy-list strings) #'> :key #'length))))
+    (if boundary
+        `(:sequence ,boundary ,alternation ,boundary)
+        alternation)))
+
+
+
+(defun make-tmlanguage-elixir ()
+  (let* ((patterns (make-tm-patterns
+                    (make-tm-match ":\\w+" 
+                                   :name 'syntax-constant-attribute) 
+                    (make-tm-match "\\w+:"
+                                   :name 'syntax-constant-attribute)
+                    (make-tm-line-comment-region "#")
+                    (make-tm-string-region "\"")
+                    (make-tm-string-region "'")
+                    (make-tm-string-region "\"\"\"")
+                    (make-tm-match (tokens :word-boundary 
+                                           (append *elixir-boolean-literals*
+                                                   *elixir-null-literal*))
+                                   :name 'syntax-constant-attribute)
+                    (make-tm-match (tokens :word-boundary *elixir-keywords*)
+                                   :name 'syntax-keyword-attribute)
+                    (make-tm-match (tokens nil *elixir-operators*)
+                                   :name 'syntax-builtin-attribute))))
+    (make-tmlanguage :patterns patterns)))
+
+(defvar *elixir-syntax-table*
+  (let ((table (make-syntax-table
+                :space-chars '(#\space #\tab #\newline)
+                :paren-pairs '((#\( . #\))
+                               (#\{ . #\})
+                               (#\[ . #\]))
+                :string-quote-chars '(#\" #\')
+                :block-string-pairs '(("\"\"\"" . "\"\"\"")
+                                      ("'''" . "'''"))
+                :line-comment-string "#"))
+        (tmlanguage (make-tmlanguage-elixir)))
+    (set-syntax-parser table tmlanguage)
+    table))
+
+(define-major-mode elixir-mode language-mode
+    (:name "elixir"
```

Comment:
I would like it to be a capital-case because it is displayed on the modeline.

Reply:
Done :+1: